pkgs/applications/editors/vscode/extensions/default.nix

```diff
@@ -2703,6 +2703,18 @@ let
 
       ms-vsliveshare.vsliveshare = callPackage ./ms-vsliveshare-vsliveshare { };
 
+      vscjava.vscode-gradle = buildVscodeMarketplaceExtension {
+        mktplcRef = {
+          name = "vscjava";
+          publisher = "vscode-gradle";
+          version = "3.12.2022110200";
```

Comment:
```suggestion
          version = "3.12.2023030400";
```

Reply:
I'm struggling to download this extension with `buildVscodeMarketplaceExtension`. If I get it from VS Code marketplace Version History tab I can download it with no issues from <https://marketplace.visualstudio.com/_apis/public/gallery/publishers/vscjava/vsextensions/vscode-gradle/3.12.2023032100/vspackage>.

```console
> nix repl '<nixpkgs>'
> :b pkgs.vscode-utils.buildVscodeMarketplaceExtension {
  mktplcRef = {
    name = "vscjava";
    publisher = "vscode-gradle";
    version = "3.12.2023032100";
    sha256 = "sha256-AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=";
  };
}

error: builder for '/nix/store/1d2paxk03rziz1bi26yida6fiwm49n3r-vscode-gradle-vscjava.zip.drv' failed with exit code 1;
       last 7 log lines:
       >
       > trying https://vscode-gradle.gallery.vsassets.io/_apis/public/gallery/publisher/vscode-gradle/extension/vscjava/3.12.2023032100/assetbyname/Microsoft.VisualStudio.Services.VSIXPackage
       >   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
       >                                  Dload  Upload   Total   Spent    Left  Speed
       >   0   319    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
       > curl: (22) The requested URL returned error: 404
       > error: cannot download vscode-gradle-vscjava.zip from any mirror
       For full logs, run 'nix log /nix/store/1d2paxk03rziz1bi26yida6fiwm49n3r-vscode-gradle-vscjava.zip.drv'.
error: 1 dependencies of derivation '/nix/store/c2v8cy7xcldr7q2nwajbgy8c28mfm5qa-vscode-extension-vscode-gradle-vscjava-3.12.2023032100.drv' failed to build
```

Reply:
@ShamrockLee Any idea?

Reply:
The VSCode extension gallery API is left undocumented. The only publicly-available "documentation" is the source code of VSCode OSS.

I guess they have changed the URL used to download the extension file (`.vsix`), and could add the new URL into the `urls` list of the fetcher.

However, I constantly got blocked:

```json
{
  "$id": "1",
  "innerException": null,
  "message": "Request was blocked due to exceeding usage of resource Count in namespace AnonymousId. For more information on why your request was blocked, see the topic \"Rate limits\" on the Microsoft Web site (https://go.microsoft.com/fwlink/?LinkId=823950).",
  "typeName": "Microsoft.TeamFoundation.Framework.Server.RequestBlockedException, Microsoft.TeamFoundation.Framework.Server",
  "typeKey": "RequestBlockedException",
  "errorCode": 0,
  "eventId": 3000
}
```

and all my attempts to use `curl` and `wget` to download from this url failed. (My knowledge toward cURL and crowlers are still limited. Maybe there's a way to satisfy the site?)

Reply:
@AmeerTaweel Have you encountered similar issues (packages like `vscode-gradle.vscjava` failed to fetch) on your [nix-vscode-extensions](https://github.com/nix-community/nix-vscode-extensions) project?

Reply:
@ShamrockLee Can't we just package extensions from sources? Looks ideal. Thoughts?

Reply:
```
vscjava.vscode-gradle = buildVscodeMarketplaceExtension rec {
        mktplcRef = {
          publisher = "vscjava";
          name = "vscode-gradle";
          version = "3.12.6";
        };
        vsix = fetchurl {
          name = "vscjava.vscode-gradle.zip";
          url = "https://github.com/microsoft/vscode-gradle/releases/download/${mktplcRef.version}/${mktplcRef.publisher}.${mktplcRef.name}-${mktplcRef.version}.vsix";
          hash = "sha256-j4JyhNGsRlJmS8Wj38gLpC1gXVvdPx10cgzP8dXmmNo=";
        };
};
```

@rhoriguchi See if this works for you.
@ShamrockLee Any improvement on format?

Reply:
Whenever available, we could make use of VSIX asset by default. What do you think?

Reply:
@superherointj Looks good by eyeball (not teted yet).

Reply:
Sorry @superherointj and @ShamrockLee I'm an idiot and can't read, for some reason I switched `name` and `publisher` while making my changes 🤦🏻‍♂️.